src/Domain/DonationRepository.php

```diff
@@ -459,6 +459,35 @@ public function findReadyToClaimGiftAid(bool $withResends): array
         return $qb->getQuery()->getResult();
     }
 
+    /**
+     * @return Donation[]
+     */
+    public function findNotFullyMatchedToCampaignsWhichClosedSince(DateTime $closedSinceDate): array
```

Comment:
I wonder if it's worth extracting the common parts of this and `findRecentNotFullyMatchedToMatchCampaigns` somehow into a private function or something so we can see what differs more easily.

Reply:
Lets look for a way to do it in future if we find we're needing to modify both queries and keep them in sync.